x-pack/plugins/siem/public/components/open_timeline/helpers.test.ts

```diff
@@ -0,0 +1,172 @@
+
+describe('helpers', () => {
+  let mockResults: TimelineResult[];
+
+  beforeEach(() => {
+    mockResults = cloneDeep(mockTimelineResults);
+  });
+
+  describe('#getPinnedEventCount', () => {
+    test('returns 6 when the timeline has 6 pinned events', () => {
+      const with6Events = mockResults[0];
+
+      expect(getPinnedEventCount(with6Events)).toEqual(6);
+    });
+
+    test('returns zero when the timeline has an empty collection of pinned events', () => {
+      const withPinnedEvents = { ...mockResults[0], pinnedEventIds: {} };
+
+      expect(getPinnedEventCount(withPinnedEvents)).toEqual(0);
+    });
+
+    test('returns zero when pinnedEventIds is undefined', () => {
+      const withPinnedEvents = omit('pinnedEventIds', { ...mockResults[0] });
+
+      expect(getPinnedEventCount(withPinnedEvents)).toEqual(0);
+    });
+
+    test('returns zero when pinnedEventIds is null', () => {
+      const withPinnedEvents = omit('pinnedEventIds', { ...mockResults[0] });
+
+      expect(getPinnedEventCount(withPinnedEvents)).toEqual(0);
+    });
+  });
+
+  describe('#getNotesCount', () => {
+    test('returns a total of 4 notes when the timeline has 4 notes (event1 [2] + event2 [1] + global [1])', () => {
+      const with4Notes = mockResults[0];
+
+      expect(getNotesCount(with4Notes)).toEqual(4);
+    });
+
+    test('returns 1 note (global [1]) when eventIdToNoteIds is undefined', () => {
+      const with1Note = omit('eventIdToNoteIds', { ...mockResults[0] });
+
+      expect(getNotesCount(with1Note)).toEqual(1);
+    });
+
+    test('returns 1 note (global [1]) when eventIdToNoteIds is null', () => {
+      const eventIdToNoteIdsIsNull = {
+        ...mockResults[0],
+        eventIdToNoteIds: null,
+      };
+      expect(getNotesCount(eventIdToNoteIdsIsNull)).toEqual(1);
+    });
+
+    test('returns 1 note (global [1]) when eventIdToNoteIds is empty', () => {
+      const eventIdToNoteIdsIsEmpty = {
+        ...mockResults[0],
+        eventIdToNoteIds: {},
+      };
+      expect(getNotesCount(eventIdToNoteIdsIsEmpty)).toEqual(1);
+    });
+
+    test('returns 3 notes (event1 [2] + event2 [1]) when noteIds is undefined', () => {
+      const noteIdsIsUndefined = omit('noteIds', { ...mockResults[0] });
+
+      expect(getNotesCount(noteIdsIsUndefined)).toEqual(3);
+    });
+
+    test('returns 3 notes (event1 [2] + event2 [1]) when noteIds is null', () => {
+      const noteIdsIsNull = {
+        ...mockResults[0],
+        noteIds: null,
+      };
+
+      expect(getNotesCount(noteIdsIsNull)).toEqual(3);
+    });
+
+    test('returns 3 notes (event1 [2] + event2 [1]) when noteIds is empty', () => {
+      const noteIdsIsEmpty = {
+        ...mockResults[0],
+        noteIds: [],
+      };
+
+      expect(getNotesCount(noteIdsIsEmpty)).toEqual(3);
+    });
+
+    test('returns 0 when eventIdToNoteIds and noteIds are undefined', () => {
+      const eventIdToNoteIdsAndNoteIdsUndefined = omit(['eventIdToNoteIds', 'noteIds'], {
```

Comment:
`omit` is deprecated and removed in 5, fyi:
https://github.com/lodash/lodash/issues/2930